client/src/components/Upload.vue

```diff
@@ -16,16 +16,30 @@ function prepareFiles(files) {
   const csvFilter = (file) => /\.csv$/i.test(file.name);
   const imageFilter = (file) => imageFilesRegEx.test(file.name);
 
-  if (files.find(videoFilter)) {
-    return [
-      'video',
-      files.filter((file) => videoFilter(file) || csvFilter(file)),
-    ];
+  const videoFiles = files.filter(videoFilter);
+  const imageFiles = files.filter(imageFilter);
+  const csvFiles = files.filter(csvFilter);
+
+  if (videoFiles.length > 0 && imageFiles.length > 0) {
+    throw new Error('Do not upload images and videos in the same batch.');
+  } else if (csvFiles.length > 1) {
+    throw new Error('Can only upload a single CSV Annotation per import');
+  } else if (videoFiles.length > 1 && csvFiles.length > 0) {
+    throw new Error('Annotation upload is not supported when multiple videos are uploaded');
+  } else if (videoFiles.length) {
+    return {
+      type: 'video',
+      media: videoFiles,
+      csv: csvFiles,
+    };
+  } else if (imageFiles.length) {
+    return {
+      type: 'image-sequence',
+      media: imageFiles,
+      csv: csvFiles,
+    };
   }
-  return [
-    'image-sequence',
-    files.filter((file) => imageFilter(file) || csvFilter(file)),
-  ];
+  throw new Error('Function should not be called on empty array');
```

Comment:
Since this is a user facing error should it be more descriptive to the end user.  Something more like : "Upload should contain images or video files."?

Reply:
This should not be a user facing error.  the method should not be called with an empty array, and there is currently no code path for doing so.

JS doesn't have typed exceptions but if it did, this would be an unhandled type.

Reply:
We don't filter the upload dialog to match the types we have listed (*.jpg, *.mp4 *.csv) and others.  If you upload a file or multiples that aren't classified as valid image or video types I believe the user would get this error.  While the array isn't empty, once it is filtered `videoFiles.length` `imageFiles.length` and `csvFiles.length` could all be zero.

![image](https://user-images.githubusercontent.com/61746913/80240045-3fc9dc00-862f-11ea-9626-286c45059404.png)

Reply:
Fixed in b2848312cc949f1d9bcd5045953c7a8b1904fd46